crates/subspace-farmer/src/farmer_cache.rs

```diff
@@ -554,21 +492,14 @@ where
             })
             .collect::<HashMap<_, _>>();
 
-        let mut piece_caches_capacity_used = vec![0u32; caches.backends.len()];
+        let mut piece_caches_capacity_used = vec![0u32; caches.backends_count()];
```

Comment:
You should already know number of backends here without calling `caches`, just store it in a variable above and use here.